File: CS/ColorEditExample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Columns;
using DevExpress.Utils;
using System.Text.RegularExpressions;

namespace ColorEditExample
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            DataTable dt = FillDataTable();
            gridControl1.DataSource = dt;
            gridControl2.DataSource = dt;
            gridView1.Columns["ColorString"].Visible = false;

            GridColumn unboundColumn = gridView1.Columns.AddField("Color");
            unboundColumn.VisibleIndex = gridView1.Columns.Count;
            unboundColumn.UnboundType = DevExpress.Data.UnboundColumnType.Object;
            RepositoryItemColorEdit ce = new RepositoryItemColorEdit();
            ce.ShowCustomColors = false;
            unboundColumn.ColumnEdit = ce;
        }

        private void gridView1_CustomUnboundColumnData(object sender, DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs e)
        {
            GridView view = sender as GridView;
            DataView dv = view.DataSource as DataView;
            if (e.IsGetData)
                e.Value = GetColorFromString(dv[e.ListSourceRowIndex]["ColorString"].ToString());
            else
                dv[e.ListSourceRowIndex]["ColorString"] = ((Color)e.Value).Name;
        }

        Color GetColorFromString(string colorString)
        {
            Color color = Color.Empty;
            ColorConverter converter = new ColorConverter();
            try
            { 
                color = (Color)converter.ConvertFromString(colorString); 
            }
            catch 
            { }
            return color;
        }

        DataTable FillDataTable()
        {
            DataTable _dataTable = new DataTable();
            DataColumn col;
            DataRow row;

            col = new DataColumn("ID", System.Type.GetType("System.Int32"));
            _dataTable.Columns.Add(col);
            col = new DataColumn("ColorString", System.Type.GetType("System.String"));
            _dataTable.Columns.Add(col);

            row = _dataTable.NewRow();
            row["ID"] = "1"; row["ColorString"] = "Red";
            _dataTable.Rows.Add(row);
            row = _dataTable.NewRow();
            row["ID"] = "2"; row["ColorString"] = "Green";
            _dataTable.Rows.Add(row);
            row = _dataTable.NewRow();
            row["ID"] = "3"; row["ColorString"] = "Blue";
            _dataTable.Rows.Add(row);

            return _dataTable;
        }
    }
}

File: VB/ColorEditExample/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Columns

Namespace ColorEditExample

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim dt As DataTable = FillDataTable()
            gridControl1.DataSource = dt
            gridControl2.DataSource = dt
            gridView1.Columns("ColorString").Visible = False
            Dim unboundColumn As GridColumn = gridView1.Columns.AddField("Color")
            unboundColumn.VisibleIndex = gridView1.Columns.Count
            unboundColumn.UnboundType = DevExpress.Data.UnboundColumnType.Object
            Dim ce As RepositoryItemColorEdit = New RepositoryItemColorEdit()
            ce.ShowCustomColors = False
            unboundColumn.ColumnEdit = ce
        End Sub

        Private Sub gridView1_CustomUnboundColumnData(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs)
            Dim view As GridView = TryCast(sender, GridView)
            Dim dv As DataView = TryCast(view.DataSource, DataView)
            If e.IsGetData Then
                e.Value = GetColorFromString(dv(e.ListSourceRowIndex)("ColorString").ToString())
            Else
                dv(e.ListSourceRowIndex)("ColorString") = CType(e.Value, Color).Name
            End If
        End Sub

        Private Function GetColorFromString(ByVal colorString As String) As Color
            Dim color As Color = Color.Empty
            Dim converter As ColorConverter = New ColorConverter()
            Try
                color = CType(converter.ConvertFromString(colorString), Color)
            Catch
            End Try

            Return color
        End Function

        Private Function FillDataTable() As DataTable
            Dim _dataTable As DataTable = New DataTable()
            Dim col As DataColumn
            Dim row As DataRow
            col = New DataColumn("ID", Type.GetType("System.Int32"))
            _dataTable.Columns.Add(col)
            col = New DataColumn("ColorString", Type.GetType("System.String"))
            _dataTable.Columns.Add(col)
            row = _dataTable.NewRow()
            row("ID") = "1"
            row("ColorString") = "Red"
            _dataTable.Rows.Add(row)
            row = _dataTable.NewRow()
            row("ID") = "2"
            row("ColorString") = "Green"
            _dataTable.Rows.Add(row)
            row = _dataTable.NewRow()
            row("ID") = "3"
            row("ColorString") = "Blue"
            _dataTable.Rows.Add(row)
            Return _dataTable
        End Function
    End Class
End Namespace
